### PR TITLE
Save the column positions

### PR DESCRIPTION
**Pull request by EstherDarkish**
Uses https://github.com/gridsuite/study-server/pull/654

### CODE REVIEW

File: src/translations/fr.json [hunk 1]
```diff
@@ -597,6 +597,7 @@
     "NodeCreateError": "Une erreur est survenue lors de la création du nœud",
     "NodeDeleteError": "Une erreur est survenue lors de la suppression du nœud",
     "NodeUpdateError": "Une erreur est survenue lors de la mise à jour du nœud",
+    "NodeUpdateColumnPositions": "Une erreur est survenue lors de la mise à jour de la positions des nœuds dans l'arbre",
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/translations/fr.json:600]: done

File: src/redux/reducer.ts [hunk 2]
```diff
@@ -120,14 +120,14 @@ import {
     NETWORK_MODIFICATION_TREE_NODE_MOVED,
     NETWORK_MODIFICATION_TREE_NODES_REMOVED,
     NETWORK_MODIFICATION_TREE_NODES_UPDATED,
-    NETWORK_MODIFICATION_TREE_SWITCH_NODES,
+    NETWORK_MODIFICATION_TREE_REORDER_NODES,
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/redux/reducer.ts:123]: ok

File: src/redux/actions.ts [hunk 3]
```diff
@@ -317,21 +317,21 @@ export function networkModificationTreeNodeMoved(
...
-    nodeToMoveId: string;
-    destinationNodeId: string;
+    parentNodeId: string;
+    nodeIds: string[];
 };
-export function networkModificationTreeSwitchNodes(
-    nodeToMoveId: string,
-    destinationNodeId: string
-): NetworkModificationTreeSwitchNodesAction {
+export function networkModificationTreeReorderNodes(
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/redux/actions.ts:327]: ok

File: src/components/network-modification-tree.jsx [hunk 4]
```diff
@@ -218,29 +220,47 @@ const NetworkModificationTree = ({
...
     /**
      * When the user stops dragging a node and releases it to its new position, we check if we need
-     * to switch the order of the moved branch with a neighboring branch.
+     * to reorder the nodes
      */
     const handleEndNodeDragging = () => {
         let movedNode = nodesMap.get(draggedBranchIdRef.current);
         draggedBranchIdRef.current = null;
-        if (movedNode) {
+        if (movedNode && movedNode.parentId) {
```
Comment (SlimaneAmar) [inline src/components/network-modification-tree.jsx:247]: `movedNode?.parentId`

**Review by SlimaneAmar — APPROVED** [review]

File: src/components/graph/network-modification-tree-model.ts [hunk 5]
```diff
@@ -30,119 +30,54 @@ export default class NetworkModificationTreeModel {
...
-        // We find the nodes from the two branches that share the same parent
-        const commonAncestor = this.getCommonAncestor(nodeToMove, destinationNode);
-        if (commonAncestor) {
-            const siblingFromNodeToMoveBranch = this.getChildOfAncestorInLineage(commonAncestor, nodeToMove);
-            const siblingFromDestinationNodeBranch = this.getChildOfAncestorInLineage(commonAncestor, destinationNode);
-            if (siblingFromNodeToMoveBranch && siblingFromDestinationNodeBranch) {
-                this.switchSiblingsOrder(siblingFromNodeToMoveBranch, siblingFromDestinationNodeBranch);
-            }
+        if (children.map((child) => child.id).join(',') === orderedNodeIds.join(',')) {
+            // Alreay in the same order.
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/graph/network-modification-tree-model.ts:59]: yes

File: src/components/graph/network-modification-tree-model.ts [hunk 6]
```diff
@@ -30,119 +30,54 @@ export default class NetworkModificationTreeModel {
...
+            return false;
         }
+        // Let's reorder the children :
+        // We create a map of children node ids and number of nodes in each of these child's subtree,
+        // then in nodeIds order, we cut and paste the corresponding number of nodes in treeNodes.
+        const justAfterParentIndex = 1 + this.treeNodes.findIndex((n) => n.id === parentNodeId); // we add 1 here to set the index just after the parent node
+        let insertedNodes = 0;
+        const nodeIdAndCorrespondingSubTreeSize = new Map(
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/graph/network-modification-tree-model.ts:67]: this code is removed

**Pull request merged**